### PR TITLE
M2-2124 Change score variable

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

**Review by IhorNesterenko99 — APPROVED** [review]

File: src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts [hunk 1]
```diff
@@ -93,3 +96,43 @@ export const getDefaultConditionalValue = (scoreId: string) => ({
   match: ConditionalLogicMatch.All,
   conditions: [{}],
 });
+
+export const getIsScoreIdVariable = (score: ActivitySettingsScore) => {
+  const { id } = score;
+  let isVariable = false;
+
+  const isMessageIncludeScoreId = (showMessage: boolean, message?: string) => {
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts:104]: could you move this function outside of another function, no need to initialize it each time

File: src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts [hunk 2]
```diff
@@ -93,3 +96,43 @@ export const getDefaultConditionalValue = (scoreId: string) => ({
...
+    if (showMessage) {
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts:105]: pls use early return `if (!showMessage) return;`

File: src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts [hunk 3]
```diff
@@ -93,3 +96,43 @@ export const getDefaultConditionalValue = (scoreId: string) => ({
...
+      isVariable = !!message?.includes(`[[${id}]]`);
+      if (isVariable) return;
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts:107]: do you really need this return ?

File: src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts [hunk 4]
```diff
@@ -93,3 +96,43 @@ export const getDefaultConditionalValue = (scoreId: string) => ({
...
+
+export const updateMessagesWithVariable = (
+  setValue: UseFormSetValue<FieldValues>,
+  name: string,
+  score: ActivitySettingsScore,
+  newScoreId: string,
+) => {
+  const { id, showMessage, message, conditionalLogic } = score;
+
+  const updateMessage = (fieldName: string, showMessage: boolean, message?: string) => {
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.utils.ts:128]: could you move this function outside of another function, no need to initialize it each time

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**